src/app/content/components/popUp/ChapterFilter.tsx

```diff
@@ -102,11 +119,106 @@ const ChapterFilter = ({
   </div>;
 };
 
+// tslint:disable-next-line: variable-name
+export const StyledDetails = styled.details`
+  width: 400px;
+  cursor: pointer;
+  border-bottom: 1px solid ${theme.color.neutral.formBorder};
+  ${theme.breakpoints.mobileSmall(css`
+    width: 100%;
+  `)}
+`;
+
+// tslint:disable-next-line: variable-name
+export const StyledSummary = styled.summary`
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+  padding: 0 1.6rem;
+  height: 4rem;
+
+  &::marker {
+    content: "";
+  }
+
+  ${ChapterTitle} {
+    margin-left: 0;
+  }
+`;
+
+// tslint:disable-next-line: variable-name
+export const StyledSectionItem = styled(PlainButton)`
+  display: flex;
+  align-items: center;
+  height: 4rem;
+  width: 100%;
+  text-align: left;
+  ${(props: { isSelected: boolean }) => {
+    if (props.isSelected) {
+      return 'color: #027eb5;';
+    }
+  }}
+
+  &:hover {
+    background-color: ${theme.color.neutral.pageBackground};
+  }
+
+  ${ChapterTitle} {
+    padding-left: 2.5rem;
+    width: 95%;
+  }
+`;
+
+// tslint:disable-next-line: variable-name
+const ChapterFilterWithToggling = (props: ChapterFilterWithTogglingProps) => {
+  const [isOpenChapterId, setIsOpenChapterId] = React.useState<string | null>(
+    props.selectedSection ? stripIdVersion(props.selectedSection.parent.id) : null
+  );
+
+  return <div className={props.className} tabIndex={-1}>
+    <Row>
+      <Column>
+        {Array.from(props.locationFilters.entries()).map(([chapterId, data]) => {
+          return <StyledDetails key={chapterId} open={isOpenChapterId === chapterId}>
+            <StyledSummary onClick={(e: React.MouseEvent) => {
+              e.preventDefault();
+              setIsOpenChapterId((current) => current === chapterId ? null : chapterId);
+            }}>
+              <ChapterTitle dangerouslySetInnerHTML={{__html: data.chapter.title}} />
+              <DownIcon rotate={isOpenChapterId === chapterId} />
+            </StyledSummary>
+            <div>
```

Comment:
i still think we should make all this one component where the sub-selections are optional, if there are sub-selections its a dropdown, if not you select the top level element. is there a problem with that approach that i'm not seeing?

Reply:
Dealing with different types of `props` with the same names is tricky, but I think I can try to unify them